(safe-)merge-queue should expose its status

briannafan opened the pull request:
#### 9c1e3b2893b0e7e790052ee964a72896091b5f90
<pre>
(safe-)merge-queue should expose its status
<a href="https://bugs.webkit.org/show_bug.cgi?id=262899">https://bugs.webkit.org/show_bug.cgi?id=262899</a>
rdar://problem/116687268

Reviewed by Jonathan Bedard.

Adds a comment with the build link to a PR labelled with safe-merge-queue once it is updated (either labelled with merge-queue or merging-blocked).

* Tools/CISupport/ews-build/steps.py:
(ValidateCommitterAndReviewer.fail_build):
(RemoveAndAddLabels.run):

Canonical link: <a href="https://commits.webkit.org/269684@main">https://commits.webkit.org/269684@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d311100bcaef936a79e807059afd4d3f56e4eccc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22749 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/414 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23835 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24656 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21055 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/23007 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/1487 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23272 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21971 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22989 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/250 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19731 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25509 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/262 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20605 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26826 "Passed tests") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20620 "Found 2 new API test failures: TestWebKitAPI.ServiceWorker.WindowClientNavigate, TestWebKitAPI.ServiceWorker.WindowClientNavigateCrossOrigin (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20850 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24676 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/293 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18128 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/22545 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/228 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5550 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/353 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/291 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (commented):
Do we have an example of this? I'm a bit concerned that the we'll end up with an unclickable link if we fail to fetch the build number property (which I suppose shouldn't happen, so maybe that's not important).

briannafan committed:
(safe-)merge-queue should expose its status
https://bugs.webkit.org/show_bug.cgi?id=262899
rdar://problem/116687268

Reviewed by Jonathan Bedard.

Adds a comment with the build link to a PR labelled with safe-merge-queue once it is updated (either labelled with merge-queue or merging-blocked).

* Tools/CISupport/ews-build/steps.py:
(ValidateCommitterAndReviewer.fail_build):
(RemoveAndAddLabels.run):

Canonical link: https://commits.webkit.org/269684@main
